File: step/stage10/ts/factorial.ts
const stdin = process.openStdin()

stdin.addListener("data", (d: Buffer) => {
    const n = Number(d.toString())
    console.log(f(1, n))
})

const f = (previousValue: number, currentCount: number): number => {
    if(currentCount === 0) {
        return previousValue
    }

    return f(previousValue*currentCount, currentCount-1)
}

File: step/stage10/go/factorial.go
package main

import "fmt"

func main() {
	var n int
	fmt.Scan(&n)
	fmt.Print(f(1, n))
}

func f(previousValue, currentCount int) int {
	if currentCount == 0 {
		return previousValue
	}

	return f(previousValue*currentCount, currentCount-1)
}
